code/modules/clothing/neck/neck.dm

```diff
@@ -148,11 +148,92 @@
 	icon_state = "petcollar"
 	item_color = "petcollar"
 	var/tagname = null
+	var/obj/item/card/id/id = null
 
 /obj/item/clothing/neck/petcollar/attack_self(mob/user)
 	tagname = copytext(sanitize(input(user, "Would you like to change the name on the tag?", "Name your new pet", "Spot") as null|text),1,MAX_NAME_LEN)
 	name = "[initial(name)] - [tagname]"
 
+/obj/item/clothing/neck/petcollar/proc/remove_id()
+	if (id)
+		if (ismob(loc))
+			var/mob/M = loc
+			M.put_in_hands(id)
```

Comment:
What if this fails?

Reply:
No idea! It didn't in testing but I should prolly add an else in or something.

Reply:
Main reason that's there by the way is cause it's yoinked from PDAcode. PDAcode has a lotta old shit.